index.html

```diff
@@ -4372,6 +4372,154 @@ <h2>Revision History</h2>
+of such a <em>proof chain</em> is to model an approval chain in a business
+process or a notary witnessing analog signatures. In such cases we CANNOT allow
+the <em>proofs</em> that are relied upon to be <strong>substituted</strong> with
+other proofs or removed.
+      </p>
+      <p>
+Let's see how a <em>proof chain</em> is constructed where the VP of Engineering
+signs off on the document first, based on the VP of engineering's signature and
+a review the CFO then signs off of the document, finally based on both the prior
+signatures and a review the CEO signs off on the document. Since others will be
+referring to the VP of engineering's signature we need to add a <em>id</em> to
+the proof options. First the VC of engineering signs the VC:
+      </p>
+      <pre><code class="language-javascript">
+{ // Rest of VC document not shown, this proof set will verify
+  &quot;proof&quot;: {&quot;id&quot;: &quot;urn:proof-0&quot;, &quot;verificationMethod&quot;: pubVPE,
+    &quot;proofValue&quot;: signature(privVPE, document)}
+}
+      </code></pre>
+      <p>
+Next, the CFO receives the document, verifies that the VP of engineering signed
+it and signs it based on a review and on the VP of Engineering signing. For this
+we need to set up the <em>proof chain</em> by indicating a dependency on the
+proof in the document just received, i.e.,
+<code>previousProof = &quot;urn:proof-0&quot;</code>. The
+processing algorithm has us &quot;bind&quot; the previous proof to the unsigned
+document. Denote the this proof by <em>proof0</em> and &quot;binding&quot; this
+information to the document by <code>document + proof0</code>:
+      </p>
+      <pre><code class="language-javascript">
+{ // Rest of VC document not shown
+  &quot;proof&quot;: [
+  {&quot;id&quot;: &quot;urn:proof-0&quot;, &quot;verificationMethod&quot;: pubVPE,
+    &quot;proofValue&quot;: signature(VP_Eng, document)},
+  {&quot;id&quot;: &quot;urn:proof-1&quot;, &quot;verificationMethod&quot;: pubCFO, &quot;previousProof&quot;: &quot;urn:proof-0&quot;,
+    &quot;proofValue&quot;: signature(privCFO, document + proof0)}]
+}
+      </code></pre>
+      <p>
+Now when the CEO validates the received the VC with the above <em>proof
+chain</em> they will check that the CFO signed based on the VP of Engineering
+signing it. First, they will check the proof/signature with
+<code>id = 0</code> against the public key of the VP of Engineering. Note that
+this proof/signature is over the original document. Next, the CEO will check
+the proof/signature <code>id = &quot;urn:proof-1&quot;</code> against the public
+key of the CFO.
+However, to make sure that the CFO signed with the document knowing, and with
+proof, that the VP of engineering signed we verify this signature/proof over
+the combination of the document and <em>proof0</em>. If validation is
+successful the CEO signs producing a proof over the document and
+<em>proof1</em> (the proof with <code>id = &quot;urn:proof-1&quot;</code>).
```

Comment:
This is a good example of why one should avoid HTML entities... I completely missed the "urn:" in the value because my eyes kept skipping past it. Took 3-4 scans to figure out what was in the text. Yes, readers don't see this but document editors do, and it makes review and finding mistakes harder.

Reply:
Does ReSpec do HTML escaping for us? Otherwise I assumed we had to. I wrote the initial text in Markdown and ran it through a processor.

Reply:
You don't need to do escaping for the vast majority of markup these days due to UTF-8 being widely supported.